### PR TITLE
[airflow/integration] DBT Cloud Integration

### PR DESCRIPTION
**Pull request by howardyoo**
airflow/integration

Related to : https://github.com/OpenLineage/OpenLineage/issues/1261

Signed-off-by: OpenLineage deploy bot <openlineage-bot-key@openlineage.io>

### Problem

Currently, there is NO OpenLineage Extractor for DBT CLOUD

Closes: #1261 

### Solution

This PR introduces new OpenLineage extractor for DBT cloud (https://www.getdbt.com/product/what-is-dbt/). The extractor uses DBT cloud hook provided by Airflow to communicate with DBT cloud via its API. For more information, please refer to #1261 .

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [x] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2022 contributors to the OpenLineage project

### CODE REVIEW

**Comment by mobuchowski** [comment]
@JDarDagran can we remove unneeded data, like `macros` from `manifest.json`?

File: integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+class DbtCloudExtractor(BaseExtractor):
+    default_schema = "public"
+
+    def __init__(self, operator):
+        super().__init__(operator)
+
+    @classmethod
+    def get_operator_classnames(cls) -> List[str]:
+        return ["DbtCloudRunJobOperator", "DbtCloudJobRunSensor"]
```
Comment (mobuchowski) [inline integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py:30]: Can we support `DbtCloudRunJobOperatorAsync` too?

Comment (JDarDagran) [reply integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py:30]: > @JDarDagran can we remove unneeded data, like `macros` from `manifest.json`?

Done.

> Can we support `DbtCloudRunJobOperatorAsync` too?

There's no such thing in Airflow provider. The only thing is `deferrable` that paired with `wait_for_termination` set to True makes operator wait asynchronously.

Comment (mobuchowski) [reply integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py:30]: @JDarDagran https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/dbt/cloud/operators/dbt.py#L12

Comment (JDarDagran) [reply integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py]: Okay, added both async operator and sensor from astronomer provider.

File: integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        with ThreadPoolExecutor() as executor:
+            running_tasks = [executor.submit(task) for task in tasks]
+            for running_task in running_tasks:
+                running_task.result()
+
+    def extract_on_complete(self, task_instance) -> Optional[TaskMetadata]:
+        task_meta_data = self.get_task_metadata()
+        operator = self.operator
+        if (
+            operator.__class__.__name__ == "DbtCloudRunJobOperator"
```
Comment (mobuchowski) [inline integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py:233]: `operator.task_type`?

Comment (mobuchowski) [reply integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py:233]: Should also check for `Async` versions I think

File: integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                    path=artifact,
+                    account_id=account_id,
+                    step=step,
+                )
+                for artifact in artifacts
+            }
+            results = await asyncio.gather(*tasks.values())
+        return {filename: result for filename, result in zip(tasks.keys(), results)}
+
+    def run_io_tasks_in_parallel(self, tasks):
```
Comment (mobuchowski) [inline integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py:223]: Do you use it anywhere?

**Review by mobuchowski — APPROVED** [review]
Approved, but please look at the few things I commented about

File: integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py (hunk 2 above)
Comment (JDarDagran) [reply integration/airflow/openlineage/airflow/extractors/dbt_cloud_extractor.py]: Changed and updated checks against async as well.

**Pull request merged**